url39209c0557_au_research_ANDS_RIFCS_API_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionV1Dot6Number1J8/src/org/ands/rifcs/base/Rights.java

```diff
@@ -0,0 +1,260 @@
+     *
+     * @param value The value to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setRightsStatement(final String value)
+        throws RIFCSException {
+        this.setRightsStatement(value, null);
+    }
+
+    /**
+     * Set the rightsStatement Value and URI.
+     *
+     * @param value The value to be set.
+     * @param rightsUri The rightsUri to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setRightsStatement(final String value,
+                                         final String rightsUri)
+        throws RIFCSException {
+        RightsInfo aRightsStatement = new RightsInfo(
+            this.newElement(Constants.ELEMENT_RIGHTS_STATEMENT));
+        aRightsStatement.setValue(value);
+        if (rightsUri != null) {
+            aRightsStatement.setRightsUri(rightsUri);
+        }
+        setRightsStatement(aRightsStatement);
+    }
+
+    /**
+     * return the rightsStatement.
+     *
+     * @return The rightsStatement
+     */
+    public final RightsInfo getRightsStatement() {
+        return this.rightsStatement;
+    }
+    /**
+     * Set the licence.
+     *
+     * @param aLicence The licence to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setLicence(final RightsTypedInfo aLicence)
+        throws RIFCSException {
+        this.licence = aLicence;
+        this.getElement().appendChild(this.licence.getElement());
+    }
+
+    /**
+     * Set the licence Value.
+     *
+     * @param value The licence value to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setLicence(final String value)
+        throws RIFCSException {
+        this.setLicence(value, null);
+    }
+
+    /**
+     * Set the licence Value and Type.
+     *
+     * @param value The licence value to be set.
+     * @param type The licence type to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setLicence(final String value,
+                                 final String type) throws RIFCSException {
+        this.setLicence(value, type, null);
+    }
+
+    /**
+     * Set the licence Value, URI and Type.
+     *
+     * @param value The licence value to be set.
+     * @param rightsUri The rightsUri to be set.
+     * @param type The licence type to be set.
+     * @throws RIFCSException A RIFCSException
+     */
+    public final void setLicence(final String value,
+                                 final String rightsUri, final String type)
+        throws RIFCSException {
+        RightsTypedInfo aLicence = new RightsTypedInfo(
+            this.newElement(Constants.ELEMENT_LICENCE));
+        aLicence.setValue(value);
+        if (rightsUri != null) {
+            aLicence.setRightsUri(rightsUri);
+        }
+        if (type != null) {
+            aLicence.setType(type);
+        }
+        setLicence(aLicence);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 147:156, 215:224.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.